### PR TITLE
feat: remove usage of GraphQLContext interface

### PR DESCRIPTION
**Pull request by samuelAndalon**
### :pencil: Description
Removing all references of `GraphQLContext` in favor of context map. 
GraphQL context map is the official approved context mechanism.

### :link: Related Issues
https://github.com/ExpediaGroup/graphql-kotlin/issues/1548

### CODE REVIEW

File: servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt [hunk 1]
```diff
@@ -16,20 +16,10 @@
...
-interface GraphQLContextFactory<out Context : GraphQLContext, Request> {
-
-    /**
-     * Generate GraphQL context based on the incoming request and the corresponding response.
-     * If no context should be generated and used in the request, return null.
-     */
-    @Deprecated(message = "The generic context object is deprecated in favor of the context map", replaceWith = ReplaceWith("generateContextMap(request)"))
-    suspend fun generateContext(request: Request): Context? = null
-
+interface GraphQLContextFactory<Request> {
```
Comment (smyrick) [inline servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt:22]: Maybe we can default to `GraphQLContext.of()` to be extra clear what the object here is

File: servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt [hunk 2]
```diff
@@ -31,15 +31,15 @@ import java.util.concurrent.CompletableFuture
...
 /**
 * Helper method to get values from a registered DataLoader.
 */
 fun <K, V> DataFetchingEnvironment.getValuesFromDataLoader(dataLoaderName: String, keys: List<K>): CompletableFuture<List<V>> {
     val loader = getDataLoader<K, V>(dataLoaderName) ?: throw MissingDataLoaderException(dataLoaderName)
-    return loader.loadMany(keys, listOf(this.getContext()))
+    return loader.loadMany(keys, listOf(this.graphQlContext))
 }
 
 /**
```
Comment (smyrick) [inline servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt:45]: Maybe we can deprecate the methods below too. We do have the useful extension methods on `GraphQLContext` but these just serve as one function call wrappers and we don't even use them in the lib either

File: servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt (hunk 1 above)
Comment (smyrick) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt:22]: Actually, we could make a MUCH more significant change and instead start using `GraphQLContext` as the type everywhere instead of `Map<*, Any>?`. 

We end up mapping back to a builder anyway in `GraphQLRequest.toExecutionInput()` so if we let everyone just call the builder themselves to modify that might be eaiser

Comment (smyrick) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt:22]: That would require everyone to migrate though, and almost certainly break existing code.Tthis is at least a breaking change but will possibly only require removal as someone could have migrated to the new `Map` methods

**Comment by smyrick** [comment]
Make sure you add the proper PR labels

File: servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt (hunk 1 above)
Comment (samuelAndalon) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt:22]: i think for simplicity it was decided to just return a `Map` and let the `GraphQLContextFactory` build the `GraphQLContext` using the map. 
i would keep it like that for simplicity and to not add another breaking change bc of the potential change of the return type

Comment (dariuszkuc) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt:22]: since it is a breaking change anyway -> lets approach it from a new user perspective, what would be a better user experience?

Comment (smyrick) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt:22]: So `DataFetchingEnvironment.getGraphQlContext` returns `GraphQLContext` not `Map<>` so I think it would sense that the factory sets an object and that is the one you get back out.

Comment (samuelAndalon) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLContextFactory.kt:25]: applied the updates to return a `GraphQLContext` object instead of a Map

File: servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt [hunk 3]
```diff
@@ -29,17 +29,25 @@ import java.util.concurrent.CompletableFuture
...
 fun <K, V> DataFetchingEnvironment.getValueFromDataLoader(dataLoaderName: String, key: K): CompletableFuture<V> {
     val loader = getDataLoader<K, V>(dataLoaderName) ?: throw MissingDataLoaderException(dataLoaderName)
-    return loader.load(key, this.getContext())
+    return loader.load(key, this.graphQlContext)
 }
 
 /**
 * Helper method to get values from a registered DataLoader.
 */
+@Deprecated(
```
Comment (smyrick) [inline servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt:44]: These methods are useful. I don't think we need to deprecate these. I was referring to the methods to get values from context directly on `DataFetchingEnvironment`, ie lines 57+ in this file

Comment (samuelAndalon) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt:44]: ahh ok, well we added those extensions bc of a feature request, where ppl stores in context using KClass, so is definetely more idiomatic do to

```kotlin
dataFetchingEnvironment.getFromContext<CoroutineScope>()
```

than

```kotlin
dataFetchingEnvironment.graphQLContext.get(CoroutineScope::class)
```

Comment (smyrick) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt:44]: But with the extension methods [here](https://github.com/ExpediaGroup/graphql-kotlin/blob/415a745254262c8757b0805c2e6a85e6fba9a485/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/extensions/GraphQLContextExtensions.kt#L26-L27) you can do this

```kotlin
dataFetchingEnvironment.graphQLContext.get<CoroutineScope>()
```

Comment (samuelAndalon) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt:44]: i still find it useful, but yes, it looks like an unneeded indirection, i can remove these in a separate PR.

Comment (smyrick) [reply servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/dataFetchingEnvironmentExtensions.kt:44]: I don't think they should be deleted just yet, maybe just deprecated, but yes that can be a separate PR

**Review by smyrick — APPROVED** [review]
